### PR TITLE
feat(examples): add moul/ulist

### PR DESCRIPTION
**Pull request by moul**
- [x] Depends on #3404 (cherry-picked)

Replaces #3402 
Related with #3321

### CODE REVIEW

**Comment by Gno2D2** [comment]
#### 🛠 PR Checks Summary
All **Automated Checks** passed. ✅

##### Manual Checks (for Reviewers):
- [ ] **IGNORE** the bot requirements for this PR (force green CI check)


<details><summary>Read More</summary>

🤖 This bot helps streamline PR reviews by verifying automated checks and providing guidance for contributors and reviewers.

##### ✅ Automated Checks (for Contributors):
 🟢 Maintainers must be able to edit this pull request ([more info](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork))


##### ☑️ Contributor Actions:
1. Fix any issues flagged by automated checks.
2. Follow the Contributor Checklist to ensure your PR is ready for review.
	- Add new tests, or document why they are unnecessary.
	- Provide clear examples/screenshots, if necessary.
	- Update documentation, if required.
	- Ensure no breaking changes, or include `BREAKING CHANGE` notes.
	- Link related issues/PRs, where applicable.

##### ☑️ Reviewer Actions:
1. Complete manual checks for the PR, including the guidelines and additional checks if applicable.

##### 📚 Resources:
- [Report a bug with the bot](https://github.com/gnolang/gno/issues/3238).
- [View the bot’s configuration file](https://github.com/gnolang/gno/tree/master/contribs/github-bot/internal/config/config.go).

<details><summary><b>Debug</b></summary><blockquote>
<details><summary><b>Automated Checks</b></summary><blockquote>

<details><summary>Maintainers must be able to edit this pull request (more info)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 The pull request was created from a fork (head branch repo: moul/gno)

```
### Then
```
🟢 Requirement satisfied
└── 🟢 Maintainer can modify this pull request

```
</blockquote></details>

</blockquote></details>


<details><summary><b>Manual Checks</b></summary><blockquote>

<details><summary>**IGNORE** the bot requirements for this PR (force green CI check)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Can be checked by

- Any user with comment edit permission

</blockquote></details>

</blockquote></details>

</blockquote></details>

</details>

File: examples/gno.land/p/moul/ulist/ulist.gno [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+	Set(index int, value interface{}) error
+
+	// Must variants that panic instead of returning errors
+	MustDelete(indices ...int)
+	MustGet(index int) interface{}
+	MustSet(index int, value interface{})
+
+	// Range operations
+	GetRange(start, end int) []Entry
+	GetRangeByOffset(offset int, count int) []Entry
```
Comment (moul) [inline examples/gno.land/p/moul/ulist/ulist.gno:430]:
```suggestion
	GetByOffset(offset int, count int) []Entry
```

Could be better, nope?

File: examples/gno.land/p/moul/ulist/ulist.gno [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+	var entries []Entry
+	l.IteratorByOffset(offset, count, func(index int, value interface{}) bool {
+		entries = append(entries, Entry{Index: index, Value: value})
+		return false
+	})
+	return entries
+}
+
+// IList defines the interface for list operations.
+// It provides methods for appending, accessing, deleting, and iterating over elements.
```
Comment (moul) [inline examples/gno.land/p/moul/ulist/ulist.gno:413]:
```suggestion
// IList defines the interface for an ulist.List compatible structure.
```

File: examples/gno.land/p/moul/ulist/ulist.gno (hunk 1 above)
Comment (moul) [reply examples/gno.land/p/moul/ulist/ulist.gno]: Done.

**Review by notJoon — APPROVED** [review]
It's a concise and perfect implementation. LGTM 👍

**Pull request merged**